include/psa/crypto_values.h

```diff
@@ -1804,6 +1933,18 @@
      (alg & PSA_ALG_AEAD_AT_LEAST_THIS_LENGTH_FLAG) != 0 :  \
      (alg) == PSA_ALG_ANY_HASH)
 
+/** Get the hash used by a composite algorithm.
+ *
+ * \param alg An algorithm identifier (value of type #psa_algorithm_t).
+ *
+ * \return The underlying hash algorithm if alg is a composite algorithm that
+ * uses a hash algorithm.
+ *
+ * \return #PSA_ALG_NONE if alg is not a composite algorithm that uses a hash.
+ */
+#define PSA_ALG_GET_HASH(alg) \
```

Comment:
If there's now a generic 'get hash component if there is one' macro, is it then expected to replace all of the other `_GET_HASH` instances?
If so, they should be moved to deprecated status.
If not, then this one should probably be named `PSA_ALG_PBKDF2_HMAC_GET_HASH` to be in-line with the other `GET_HASH`'s

Reply:
[`PSA_ALG_GET_HASH`](https://armmbed.github.io/mbed-crypto/html/api/ops/algorithms.html?highlight=psa_alg_get_hash#c.PSA_ALG_GET_HASH) is in the 1.0.0 spec. Adding it is part of https://github.com/ARMmbed/mbedtls/issues/3278. We've decided to add it now so that we wouldn't need to introduce a `PSA_ALG_PBKDF2_HMAC_GET_HASH` only to deprecate it soon afterwards.

Reply:
OK, thanks for the explanation.